src/vellum/workflows/resolvers/resolver.py

```diff
@@ -16,27 +18,66 @@ def get_latest_execution_events(self) -> Iterator[WorkflowEvent]:
     def get_state_snapshot_history(self) -> Iterator[BaseState]:
         return iter([])
 
-    def load_state(self, previous_execution_id: Optional[UUID] = None) -> Optional[BaseState]:
+    def _find_execution_events(
+        self, execution_id: str, spans: list
+    ) -> Tuple[Optional[WorkflowExecutionInitiatedEvent], Optional[WorkflowExecutionInitiatedEvent]]:
+        previous_invocation: Optional[WorkflowExecutionInitiatedEvent] = None
+        root_invocation: Optional[WorkflowExecutionInitiatedEvent] = None
+
+        for span in spans:
+            # Look for workflow execution spans
+            if isinstance(span, WorkflowExecutionSpan):
```

Comment:
[nit] we typically use `.name` and `.type` attribute references to type discriminate